mcu/src/mcu/uart_reader.py

```diff
@@ -0,0 +1,455 @@
+        while not self.stop_thread.is_set():
+            if self.serial_connection and self.serial_connection.is_open:
+                try:
+                    line = self.serial_connection.readline().decode("ascii").strip()
+                    if line:
+                        buffer = self.parse_buffer(line)
+                        if buffer is not None:
+                            dt = np.dtype(np.uint16)
+                            dt = dt.newbyteorder("<")
+                            buffer_array = np.frombuffer(buffer, dtype=dt)
+                            self.print_stuff(f"Aquisition Number \t: {msg_counter}")
+                            self.print_stuff(
+                                f">Aquisition Memory\t: {len(line) - len(PRINT_PREFIX)} bytes"
+                            )
+                            self.print_stuff(
+                                f">Aquisition Size  \t: {len(buffer_array)} samples"
+                            )
+                            plot_signal_and_fft(buffer_array, FREQ_SAMPLING)
+                except Exception as e:
+                    self.print_stuff(f"Error: {e}", "error")
+                    self.disconnect()
+                    return
+            else:
+                break
+
+    def parse_buffer(self, line):
+        if line.startswith(PRINT_PREFIX):
+            return bytes.fromhex(line[len(PRINT_PREFIX) :])
+        else:
+            self.print_stuff(line)
+            return None
+
+    def on_closing(self):
+        self.stop_thread.set()
+        if self.serial_thread and self.serial_thread.is_alive():
+            self.serial_thread.join(timeout=1)
+        if self.serial_connection and self.serial_connection.is_open:
+            try:
+                self.serial_connection.close()
+            except Exception as e:
+                self.print_stuff(f"Error closing connection: {e}", "error")
+        self.root.destroy()
+
+
+def launch_gui():
+    root = tk.Tk()
+    app = UARTReaderApp(root)
+    root.protocol("WM_DELETE_WINDOW", app.on_closing)
+    root.mainloop()
+
+
+stop_event = threading.Event()
+
+
+def signal_handler(sig, frame):
+    print("Interrupt received, stopping...")
+    stop_event.set()
+
+
+@click.command()
+@click.option(
+    "-p", "--port", default=None, type=str, help="Port for serial communication"
+)
+@click.option(
+    "-b",
+    "--baudrate",
+    default=DEFAULT_BAUDRATE,
+    type=int,
+    help="Baudrate for serial communication",
+)
+@click.option(
+    "-f", "--freq-sampling", default=FREQ_SAMPLING, type=int, help="Sampling frequency"
+)
+@click.option("-m", "--max-adc", default=VAL_MAX_ADC, type=int, help="Max ADC value")
+@click.option("-v", "--vdd", default=VDD, type=float, help="VDD value")
+@click.option(
+    "-l", "--log", is_flag=True, type=bool, default=False, help="Enable logging"
+)
+@click.option(
+    "-d",
+    "--auto-delete",
+    is_flag=True,
+    type=bool,
+    default=False,
+    help="Auto delete audio files",
```

Comment:
In general, this is more common to have a `-o/--overwrite` option (or `-f/--force`).